GitUI/CommandsDialogs/FormResolveConflicts.cs

```diff
@@ -874,38 +894,115 @@ private void ChooseRemoteOnConflict(string fileName)
+                Text = text,
+                InstructionText = instructionText,
+                Caption = caption,
+                StandardButtons = TaskDialogStandardButtons.Cancel,
+                Icon = TaskDialogStandardIcon.Error,
+                FooterCheckBoxText = applyToAllCheckBoxText,
+                FooterIcon = TaskDialogStandardIcon.Information,
+                StartupLocation = TaskDialogStartupLocation.CenterOwner,
+                Cancelable = true
+            };
 
-            using (var frm = new FormModifiedDeletedCreated(string.Format(_chooseLocalButtonText.Text + " ({0})", GetLocalSideString()),
-                                                                            string.Format(_chooseRemoteButtonText.Text + " ({0})", GetRemoteSideString()),
-                                                                            _keepBaseButtonText.Text,
-                                                                            caption))
+            // Local
+            var btnKeepLocal = new TaskDialogCommandLink("KeepLocal", null, keepLocalButtonText);
+            btnKeepLocal.Click += (s, e) =>
             {
-                frm.ShowDialog(this);
+                _solveMergeConflictDialogResult = SolveMergeConflictActionType.atKeepLocal;
+                dialog.Close();
+            };
 
-                if (frm.KeepBase)
-                {
-                    // base
-                    ChooseBaseOnConflict(item.Filename);
-                }
+            // Remote
+            var btnKeepRemote = new TaskDialogCommandLink("KeepRemote", null, keepRemoteButtonText);
+            btnKeepRemote.Click += (s, e) =>
+            {
+                _solveMergeConflictDialogResult = SolveMergeConflictActionType.atKeepRemote;
+                dialog.Close();
+            };
 
-                if (frm.KeepLocal)
-                {
-                    // local
-                    ChooseLocalOnConflict(item.Filename);
-                }
+            // Base
+            var btnKeepBase = new TaskDialogCommandLink("KeepBase", null, keepBaseButtonText);
+            btnKeepBase.Click += (s, e) =>
+            {
+                _solveMergeConflictDialogResult = SolveMergeConflictActionType.atKeepBase;
+                dialog.Close();
+            };
+
+            dialog.Controls.Add(btnKeepLocal);
+            dialog.Controls.Add(btnKeepRemote);
+            dialog.Controls.Add(btnKeepBase);
+
+            return dialog;
+        }
+
+        private void OpenSolveMergeConflictDialogAndExecuteSelectedMergeAction(SolveMergeConflictCheckRevisionDelegate selectedMergeAction,
+            string dialogText, string dialogInstructionText, string dialogCaption, string dialogFooterCheckboxText,
+            string keepLocalButtonText, string keepRemoteButtonText, string keepBaseButtonText)
+        {
+            bool shouldOpenDialog = _SolveMergeConflictApplyToAll == false;
+            if (shouldOpenDialog)
```

Comment:
Non nullable bools don't need this notation. The same applies below to other bool operations.

The temp variable is not required:
```suggestion
            if (!_solveMergeConflictApplyToAll)
```